libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -159,25 +189,75 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
     options: SourceOptions
   ): SetConfigurationSettingParam<FeatureFlagValue>[] {
     const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+    const featureFlags = new Array<FeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.foundMsFeatureManagement) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaSection(config);
+
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} is in an invalid format: id is a required property`
```

Comment:
It's not necessary to print the entire ff.

Feature flag without id is found, id is a required property.

Reply:
Updated